src/bid/bid.rs

```diff
@@ -0,0 +1,102 @@
+//! Bid data structure
+//!
+
+use crate::score_gen::score::compute_score;
+use dusk_bls12_381::Scalar;
+use jubjub::{AffinePoint, Scalar as JubJubScalar};
+use poseidon252::sponge::sponge::sponge_hash;
+
+#[derive(Copy, Clone, Debug)]
+pub struct Bid {
+    // Pub Inputs
+    //
+    // B^R
+    pub(crate) bid_tree_root: Scalar,
+    // sigma^s
+    pub(crate) consensus_round_seed: Scalar,
+    // k^r
+    pub(crate) latest_consensus_round: Scalar,
+    // k^s
+    pub(crate) latest_consensus_step: Scalar,
+    //
+    // Public Outputs
+    //
+    // i (One time identity of the prover)
+    pub(crate) prover_id: Option<Scalar>,
+    // q (Score of the bid)
+    pub(crate) score: Option<Scalar>,
+    //
+    // Private Inputs
+    //
+    // v
+    pub(crate) value: JubJubScalar,
+    // r
+    pub(crate) randomness: JubJubScalar,
+    // k
+    pub(crate) secret_k: Scalar,
+    // R = r * G
+    pub(crate) pk: AffinePoint,
+}
+
+impl Default for Bid {
```

Comment:
Is there a reason not to just #[derive(Default)] here?

Reply:
Nope, you're right. 
Changed!